packages/server/src/fhir/operations/ai.ts

```diff
@@ -161,11 +295,19 @@ export async function callAI(
     body: JSON.stringify(requestBody),
   });
 
+  // For streaming, return raw response
+  if (stream) {
+    return response;
+  }
+
+  // For non-streaming, parse and return structured data
   if (!response.ok) {
     const errorData = await response.json().catch(() => ({}));
-    throw new Error(
+    const error: any = new Error(
```

Comment:
Instead of casting to any here, can we instead at line 309 do a type assertion. Something like:
`(error as Error & { statusCode: number }).statusCode = response.status`